network/handlers/drain.go

```diff
@@ -56,6 +60,10 @@ type Drainer struct {
 	// Mutex guards the initialization and resets of the timer
 	sync.RWMutex
 
+	// CheckOption is the optional checker
+	// for kubelet probe to perform when the handler is not in Drain status.
+	CheckOption CheckOption
```

Comment:
```suggestion
	// HealthCheck is an optional health check that is performed until the drain signal is received.
	// When unspecified, a "200 OK" is returned, otherwise this function is invoked.
	HealthCheck http.HandlerFunc
```

Reply:
done

Reply:
Personally, I am in favor of custom name too. Do you mind if I change back to `CheckOption` and keep the comment you suggested?

Reply:
I don't understand the value we're getting from the type alias?  inlined, I know it's a well known handler interface I need to implement.  outlined I need to look it up before doing the same.

We're not defining our own methods on the type alias, or "constants" we expect to be passed as values.

Reply:
Thanks, that sounds reasonable to me. I'll keep the current change you suggested, and remove the hold.